### PR TITLE
Basic PR management

### PR DESCRIPTION
**Pull request by MarconZet**

### CODE REVIEW

File: app/src/main/kotlin/org/virtuslab/bazelsteward/app/App.kt [hunk 1]
```diff
@@ -51,14 +57,24 @@ class App(private val ctx: Context) {
...
-        ctx.gitOperations.createBranchWithChange(change)
-        if (ctx.config.pushToRemote) {
-          ctx.gitOperations.pushBranchToOrigin(branch)
-          ctx.gitHostClient.openNewPR(branch)
+      val prStatus = ctx.gitHostClient.checkPrStatus(branch)
+      when (prStatus) {
+        NONE, OPEN_NOT_MERGEABLE -> {
+          logger.info { "Creating branch ${branch.name}" }
+          runCatching {
+            ctx.gitOperations.createBranchWithChange(change)
```
Comment (lukaszwawrzyk) [inline app/src/main/kotlin/org/virtuslab/bazelsteward/app/App.kt:65]: if PR is open but non mergable, isn't there a chance that branch will exist locally and checkout will not work?

File: e2e/src/test/kotlin/org/virtuslab/bazelsteward/e2e/CountingGitHostClient.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import org.virtuslab.bazelsteward.core.GitHostClient
+
+abstract class CountingGitHostClient : GitHostClient {
+  val openNewPrCalls: MutableList<GitBranch> = ArrayList(10)
+  val closeOldPrsCalls: MutableList<GitBranch> = ArrayList(10)
+  override fun openNewPR(branch: GitBranch) {
+    openNewPrCalls.add(branch)
+  }
+
+  override fun closeOldPrs(newBranch: GitBranch) {
```
Comment (lukaszwawrzyk) [inline e2e/src/test/kotlin/org/virtuslab/bazelsteward/e2e/CountingGitHostClient.kt:13]: why is this parameter named newBranch?

File: github/src/main/kotlin/org/virtuslab/bazelsteward/github/GithubClient.kt [hunk 3]
```diff
@@ -20,23 +22,47 @@ class GithubClient private constructor(private val config: Config, repository: S
...
 
-    return true
+  private fun checkPrStatus(pr: GHPullRequest?): PrStatus {
+    return if (pr == null)
+      PrStatus.NONE
+    else if (pr.isMerged)
+      PrStatus.MERGED
+    else if (pr.state == GHIssueState.CLOSED)
+      PrStatus.CLOSED
+    else if (pr.listCommits().toList().any { it.commit.author.name != "github-actions[bot]" })
```
Comment (lukaszwawrzyk) [inline github/src/main/kotlin/org/virtuslab/bazelsteward/github/GithubClient.kt:60]: open an issue to read author from git config first or just fix it in this PR

File: github/test/kotlin/org/virtuslab/bazelsteward/github/GithubTest.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      "bazel-steward/testing/closed" to PrStatus.CLOSED,
+      "bazel-steward/testing/mergable" to PrStatus.OPEN_MERGEABLE,
+      "bazel-steward/testing/merged" to PrStatus.MERGED,
+      "bazel-steward/testing/modified" to PrStatus.OPEN_MODIFIED,
+      "bazel-steward/testing/not_mergeable" to PrStatus.OPEN_NOT_MERGEABLE,
+      "bazel-steward/testing/none" to PrStatus.NONE
+    )
+
+    val env = Environment.system
+    val config = Config(Path.of("."), false, "base")
```
Comment (lukaszwawrzyk) [inline github/test/kotlin/org/virtuslab/bazelsteward/github/GithubTest.kt:37]: maybe use named arguments here for clarity

**Review by lukaszwawrzyk — APPROVED** [review]

File: e2e/src/test/kotlin/org/virtuslab/bazelsteward/e2e/CountingGitHostClient.kt (hunk 2 above)
Comment (MarconZet) [reply e2e/src/test/kotlin/org/virtuslab/bazelsteward/e2e/CountingGitHostClient.kt:13]: The idea is that all branches that are related to the branch from the parameter are closed.

On second thought it should be changed

**Pull request merged**